src/Uri.php

```diff
@@ -262,10 +274,6 @@ public function getScheme()
 
     public function getAuthority()
     {
-        if ($this->host == '') {
-            return '';
```

Comment:
According to RFC 3986 the authority can actually have a userinfo or port without a host.
But `parse_url` cannot parse such thing. This is one reason why it's best to just define a default host for absolute http and https URIs. This circumvents this dilemma. Also it makes sure that `(new Uri())->withScheme('http')` is still a valid URI because according to RFC 7230 Section 2.7 a host is required.